tests/core/test_datamodules.py

```diff
@@ -533,6 +535,81 @@ def test_simple_hyperparameters_saving():
     assert data.hparams == AttributeDict({"arg0": 10, "arg1": "foo", "kwarg0": "bar"})
 
 
+def test_define_as_dataclass():
+    """
+    Test datamodule for initialization as Python dataclass.
+    """
+
+    class RandomDataset(torch.utils.data.Dataset):
+        def __init__(self, size, length):
+            self.len = length
+            self.data = torch.randn(length, size)
+
+        def __getitem__(self, index):
+            return self.data[index]
+
+        def __len__(self):
+            return self.len
+
+    @dataclass(init=True, repr=True, eq=True, order=True, unsafe_hash=True, frozen=False)
```

Comment:
these settings are not necessary, right?
we only have to add @dataclass, to enable the basic usage, right?

Reply:
Right. The first three parameters default to True, the other three to False. Only using @dataclass is enough. Have added all the parameters just to make sure, nothing breaks the functionality when enabled. Will remove for `BoringDataModule2`, and let it be for `BoringDataModule1`.

Reply:
Let's keep them to be sure it doesn't break. 
Anyway we can test it is actually frozen ?

Reply:
We can test that, but enabling frozen will break the functionality of the base class.

Reply:
`init` seem sunnecessary

<img width="655" alt="Screenshot 2021-08-30 at 16 41 07" src="https://user-images.githubusercontent.com/11067892/131357202-dbeb955f-cbca-482d-96e9-b7580c95d76d.png">

Reply:
Thanks for pointing the redundancy. Ideally, it should be `__post_init__`. And the other parameters are to test all the features of `dataclass` that can be enabled without breaking any functionality.